kernel/kernel-api/src/test/scala/io/delta/kernel/internal/util/SchemaUtilsSuite.scala

```diff
@@ -310,6 +311,140 @@ class SchemaUtilsSuite extends AnyFunSuite {
+      2L -> new StructField("data", StringType.STRING, true)).map { case (k, v) =>
+      java.lang.Long.valueOf(k) -> v
+    }.asJava
+
+    val schemaChanges = SchemaUtils.computeSchemaChangesById(fieldMappingBefore, fieldMappingAfter)
+
+    assert(schemaChanges.removedFields().isEmpty)
+    assert(schemaChanges.updatedFields().isEmpty)
+    assert(schemaChanges.addedFields().size() == 1)
+    assert(schemaChanges.addedFields().get(0) == fieldMappingAfter.get(2))
+  }
+
+  test("Compute schema changes with renamed fields") {
+    val fieldMappingBefore = Map(
+      1L -> new StructField("id", IntegerType.INTEGER, true)).map { case (k, v) =>
+      java.lang.Long.valueOf(k) -> v
+    }.asJava
+
+    val fieldMappingAfter = Map(
+      1L -> new StructField("renamed_id", IntegerType.INTEGER, true)).map { case (k, v) =>
+      java.lang.Long.valueOf(k) -> v
+    }.asJava
+
+    val schemaChanges = SchemaUtils.computeSchemaChangesById(fieldMappingBefore, fieldMappingAfter)
+
+    assert(schemaChanges.addedFields().isEmpty)
+    assert(schemaChanges.removedFields().isEmpty)
+    assert(schemaChanges.updatedFields().size() == 1)
+    assert(schemaChanges.updatedFields().get(0) ==
+      new Tuple2(fieldMappingBefore.get(1L), fieldMappingAfter.get(1)))
+  }
+
+  test("Compute schema changes with type changed columns") {
+    val fieldMappingBefore = Map(
+      1L -> new StructField("id", IntegerType.INTEGER, true)).map { case (k, v) =>
+      java.lang.Long.valueOf(k) -> v
+    }.asJava
+
+    val fieldMappingAfter = Map(
+      1L -> new StructField("promoted_to_long", LongType.LONG, true)).map { case (k, v) =>
+      java.lang.Long.valueOf(k) -> v
+    }.asJava
+
+    val schemaChanges = SchemaUtils.computeSchemaChangesById(fieldMappingBefore, fieldMappingAfter)
+
+    assert(schemaChanges.addedFields().isEmpty)
+    assert(schemaChanges.removedFields().isEmpty)
+    assert(schemaChanges.updatedFields().size() == 1)
+    assert(schemaChanges.updatedFields().get(0) ==
+      new Tuple2(fieldMappingBefore.get(1L), fieldMappingAfter.get(1)))
+  }
+
+  test("Compute schema changes with dropped fields") {
+    val fieldMappingBefore = Map(
+      1L -> new StructField("id", IntegerType.INTEGER, true),
+      2L -> new StructField("data", StringType.STRING, true)).map { case (k, v) =>
+      java.lang.Long.valueOf(k) -> v
+    }.asJava
+
+    val fieldMappingAfter = Map(
+      2L -> new StructField("data", StringType.STRING, true)).map { case (k, v) =>
+      java.lang.Long.valueOf(k) -> v
+    }.asJava
+
+    val schemaChanges = SchemaUtils.computeSchemaChangesById(fieldMappingBefore, fieldMappingAfter)
+
+    assert(schemaChanges.addedFields().isEmpty)
+    assert(schemaChanges.updatedFields().isEmpty)
+    assert(schemaChanges.removedFields().size() == 1)
+    assert(schemaChanges.removedFields().get(0) == fieldMappingBefore.get(1))
+  }
+
+  test("Compute schema changes with nullability change") {
```

Comment:
Can we add a test case for metadata changes?

Reply:
Done!